src/helpers/githubApi.ts

```diff
@@ -0,0 +1,34 @@
+let githubToken = '';
+
+export const saveToken = (token: string) => {
+  githubToken = token;
+  localStorage.setItem('github_token', token);
+};
+
+export const getToken = () => {
+  if (!githubToken) {
+    githubToken = localStorage.getItem('github_token') || '';
+  }
+  return githubToken;
+};
+
+export const githubRequest = async (endpoint: string, options: RequestInit = {}) => {
+  const token = getToken();
+  if (!token) {
+    throw new Error('GitHub Token 未设置');
+  }
+
+  const response = await fetch(`https://api.github.com${endpoint}`, {
+    ...options,
+    headers: {
+      ...options.headers,
+      Authorization: `Bearer ${token}`,
+    },
+  });
+
+  if (!response.ok) {
+    throw new Error(`GitHub 请求失败: ${response.statusText}`);
+  }
+
+  return response.json();
+};
```

Comment:
Can we implement this method in this way:
1. if (!gettoken) return null
2. try fetch and return whatever response and catch and throw unexpected errors.

Something like this: (not tested)
```
export const githubRequest = async (endpoint: string, options: RequestInit = {}) => {
  if (!getToken()) {
    return null
  }

try{
  const response = await fetch(`https://api.github.com${endpoint}`, {
    ...options,
    headers: {
      ...options.headers,
      Authorization: `Bearer ${token}`,
    },
  });
  return response.json();
}
catch{
    return null;
}
};
```

Things I tried to do here:
1. This function only does one thing: make github request.
2. So it only returns response if request succeded or null if no request is made.
3. We don't care about if the response is ok or error in here, whatever code calls this request should handle that. For here, we only make sure the output is valid or null.

Let me know if this makes sense.